pkg/preparer/orchestrate.go

```diff
@@ -155,7 +155,39 @@ func (p *Preparer) verifySignature(manifest pods.PodManifest, logger logging.Log
 
 	signer, err := manifest.Signer(p.keyring)
 	if signer != nil {
-		logger.WithField("signer_key", signer.PrimaryKey.KeyIdShortString()).Debugln("Resolved manifest signature")
+		signerId := signer.PrimaryKey.KeyIdShortString()
+		logger.WithField("signer_key", signerId).Debugln("Resolved manifest signature")
+
+		// Hmm, some hacks here.
+		if manifest.Id == "p2-preparer" {
+			raw, ok := manifest.Config["authorized_deployers"]
+
+			if !ok {
+				logger.NoFields().Errorln("authorized_deployers not found in preparer's config")
+				return false
+			}
+
+			authorizedDeployers, ok := raw.([]string)
```

Comment:
I have definitely had bad experiences type-asserting through slices in the past. lemme review the laws of reflection and take another look